src/widget/systemtrayicon.cpp

```diff
@@ -116,6 +113,26 @@ QString SystemTrayIcon::extractIconToFile(QIcon icon, QString name)
     return iconPath;
 }
 
+#if defined(ENABLE_SYSTRAY_GTK_BACKEND) || defined(ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND)
+GdkPixbuf* SystemTrayIcon::convertQIconToPixbuf(const QIcon &icon)
+{
+    void (*callbackFreeImage)(guchar*, gpointer) =
+        [](guchar* image_bytes, gpointer)
+    {
+        delete[] image_bytes;
```

Comment:
No garbage collection magic. Qt relies on clever smart pointers and copy-on-write with most of it's objects. Not sure about the closure though. Might leave a mem-leak on the function pointer. Hm… if you have info about the life-span of lambda assignments I'd be interested in that as well.